source/extensions/common/wasm/v8/v8.cc

```diff
@@ -0,0 +1,721 @@
+    if (left[i]->kind() != right[i]->kind()) {
+      return false;
+    }
+  }
+  return true;
+}
+
+static uint32_t parseVarint(const byte_t*& pos, const byte_t* end) {
+  uint32_t n = 0;
+  uint32_t shift = 0;
+  byte_t b;
+  do {
+    if (pos + 1 > end) {
+      throw WasmVmException("Failed to parse corrupted WASM module");
+    }
+    b = *pos++;
+    n += (b & 0x7f) << shift;
+    shift += 7;
+  } while ((b & 0x80) != 0);
+  return n;
+}
+
+// Template magic.
+
+template <typename T> struct ConvertWordType {
+  using type = T; // NOLINT(readability-identifier-naming)
+};
+template <> struct ConvertWordType<Word> {
+  using type = uint32_t; // NOLINT(readability-identifier-naming)
+};
+
+template <typename T> wasm::Val makeVal(T t) { return wasm::Val::make(t); }
+template <> wasm::Val makeVal(Word t) { return wasm::Val::make(static_cast<uint32_t>(t.u64_)); }
+
+template <typename T> constexpr auto convertArgToValKind();
+template <> constexpr auto convertArgToValKind<Word>() { return wasm::I32; };
+template <> constexpr auto convertArgToValKind<int32_t>() { return wasm::I32; };
+template <> constexpr auto convertArgToValKind<uint32_t>() { return wasm::I32; };
+template <> constexpr auto convertArgToValKind<int64_t>() { return wasm::I64; };
+template <> constexpr auto convertArgToValKind<uint64_t>() { return wasm::I64; };
+template <> constexpr auto convertArgToValKind<float>() { return wasm::F32; };
+template <> constexpr auto convertArgToValKind<double>() { return wasm::F64; };
+
+template <typename T> struct V8ProxyForGlobal : Global<T> {
+  V8ProxyForGlobal(wasm::Global* value) : global_(value) {}
+
+  T get() override { return global_->get().get<typename ConvertWordTypeToUint32<T>::type>(); };
+  void set(const T& value) override { global_->set(makeVal(static_cast<T>(value))); };
+
+  wasm::Global* global_;
+};
+
+template <typename T, std::size_t... I>
+constexpr auto convertArgsTupleToValTypesImpl(absl::index_sequence<I...>) {
+  return wasm::ownvec<wasm::ValType>::make(
+      wasm::ValType::make(convertArgToValKind<typename std::tuple_element<I, T>::type>())...);
+}
+
+template <typename T> constexpr auto convertArgsTupleToValTypes() {
+  return convertArgsTupleToValTypesImpl<T>(absl::make_index_sequence<std::tuple_size<T>::value>());
+}
+
+template <typename T, typename U, std::size_t... I>
+constexpr T convertValTypesToArgsTupleImpl(const U& arr, absl::index_sequence<I...>) {
+  return std::make_tuple(
+      (arr[I]
+           .template get<
+               typename ConvertWordType<typename std::tuple_element<I, T>::type>::type>())...);
+}
+
+template <typename T, typename U> constexpr T convertValTypesToArgsTuple(const U& arr) {
+  return convertValTypesToArgsTupleImpl<T>(arr,
+                                           absl::make_index_sequence<std::tuple_size<T>::value>());
+}
+
+// V8 implementation.
+
+bool V8::load(const std::string& code, bool /* allow_precompiled */) {
+  ENVOY_LOG(trace, "[wasm] load()");
```

Comment:
WasmVm is already having logger with Logger::Id::wasm, no need `[wasm]` in all logger call?

Reply:
Removed.